rabbitmq_amqp_python_client/connection.py

```diff
@@ -141,13 +147,25 @@ def dial(self) -> None:
                     client_key,
                     password,
                 )
-        self._conn = BlockingConnection(
-            url=self._addr,
-            urls=self._addrs,
-            ssl_domain=self._ssl_domain,
-            on_disconnection_handler=self._on_disconnection_handler,
-        )
-        self._open()
+
+        if (
```

Comment:
Can you move this code inside `_on_disconnection` to avoid this duplication code and add a log ( In case is null or `active_recovery==false` like log("recovery connection is not active the connection won't be recovered")
to be like the other clients